src/Mod/PartDesign/Gui/TaskHelixParameters.cpp

```diff
@@ -202,6 +202,7 @@ void TaskHelixParameters::fillAxisCombo(bool forceRefill)
         PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
         Part::Part2DObject* pcSketch = dynamic_cast<Part::Part2DObject*>(pcFeat->Profile.getValue());
         if (pcSketch){
+            addAxisToCombo(pcSketch,"N_Axis",QObject::tr("Normal sketch axis"));
```

Comment:
Shall also be added in retranslateUI()

Reply:
Can you please give a bit more details on this? It's my first contact with FreeCAD's source and I'm still learning. My previous C++/Qt experiences are [long ago](https://github.com/KDE/krusader).

Do you mean adding this "Normal sketch axis" entry to the respective combobox in the TaskHelixParameters.ui file? What effect does it have and how can I verify it?

Reply:
In the same file there is a function named 'retranslateUi()' that is called on language change. UI items shall also be there. ;)

Reply:
I can only find this reference
```cpp
void TaskHelixParameters::changeEvent(QEvent *e)
{
    TaskBox::changeEvent(e);
    if (e->type() == QEvent::LanguageChange) {
        ui->retranslateUi(proxy);
    }
}
```
which made me come up the question about the .ui file, where the class behind the `ui` pointer is generated from.

Reply:
@jbaehr Yes sorry, I messed a bit things up. Indeed it shall be added in the UI file so it is added in the retranslateUi() function at compile time. ;)

Reply:
done

Reply:
It's pointless to add any items to the combo box in the .ui file because when opening the helix dialog the function fillAxisCombo() is called that clears the combo box and re-adds the items. At runtime when selecting a reference axis further items might be added.

The function changeEvent() is not fully implemented to really support a language change. At the moment when you change the language in the preferences while the helix dialog is open it will clear the combo box and only re-adds the items of the .ui file. However, all items that are added by reference axes will be gone and this can cause serious problems because then the lengths of the combo box and the internal array `axesInList` don't match any more.

If you e.g. select an edge of the model as reference axis, change the language and select the same reference axis again then the current index of the combo box is set to -1 because it cannot find the item any more. Now if -1 is used to access the `axesInList` a segmentation fault is the consequence.

In order to fully support a language change a few things must be done:
* save the current index of the combo box
* save the items that come from reference axes
* call retranslateUi()
* restore the items
* restore the current index

Reply:
@wwmayer my whole mistake. Thanks for jumping in.
@jbaehr sorry for the useless noise on this

Reply:
41efee1b4c implements the language change. However, when the user has selected several reference axes then maximum one of them can be preserved. Trying to preserve all of them would lead to ugly code and since this issue is anyway quite uncommon I think we can leave like this.

Reply:
Thank you.